timelock-impl/src/main/java/com/palantir/atlasdb/timelock/paxos/BatchPingableLeaderResource.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.paxos;
+
+import java.util.Objects;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import com.palantir.common.streams.KeyedStream;
+import com.palantir.paxos.PaxosLearner;
+import com.palantir.paxos.PaxosValue;
+
+public class BatchPingableLeaderResource implements BatchPingableLeader {
+
+    private final UUID leadershipUuid;
+    private final PaxosComponents components;
+
+    public BatchPingableLeaderResource(UUID leadershipUuid, PaxosComponents components) {
+        this.leadershipUuid = leadershipUuid;
+        this.components = components;
+    }
+
+    @Override
+    public Set<Client> ping(Set<Client> clients) {
+        return KeyedStream.of(clients)
+                .map(components::learner)
+                .map(PaxosLearner::getGreatestLearnedValue)
+                .filter(Objects::nonNull)
+                .filter(this::isThisNodeTheLeaderFor)
+                .keys()
+                .collect(Collectors.toSet());
```

Comment:
Yep, this makes sense. We might want to be a bit cautious about perf at some point since this seems like it might be a semi-hot code path, but this should be fine for now

Reply:
we could do something with the acceptor cache? where the cache also contains uuid, and it could just be a thing of cache keys?

Reply:
I think that would be next steps if anything.